Src/DependencyCollector/Net45.Tests/DependencyTrackingTelemetryModuleHttpTest.cs

```diff
@@ -345,6 +344,196 @@ public void OnBeginOnEndAreNotCalledForAppInsightsUrl()
+            }
+        }
+
+        [TestMethod]
+        [Timeout(500000)]
+        public void TestDependencyCollectorPostRequestsAreCollectedEventSource()
+        {
+            using (var module = new DependencyTrackingTelemetryModule())
+            {
+                module.DisableDiagnosticSourceInstrumentation = true;
+                module.ProfileQueryEndpoint = FakeProfileApiEndpoint;
+                module.Initialize(this.config);
+
+                var url = new Uri(LocalhostUrl);
+                HttpWebRequest request = WebRequest.CreateHttp(url);
+                request.Method = "POST";
+                request.ContentLength = 1;
+
+                using (new LocalServer(LocalhostUrl))
+                {
+                    using (var stream = request.GetRequestStream())
+                    {
+                        stream.Write(new byte[1], 0, 1);
+                    }
+
+                    using (request.GetResponse())
+                    {
+                    }
+                }
+
+                this.ValidateTelemetryForEventSource(this.sentTelemetry.Single(), url, request, true, "200");
+            }
+        }
+
+        [TestMethod]
+        [Timeout(5000)]
+        public void TestHttpRequestsWithQueryStringAreCollectedDiagnosticSource()
+        {
+            using (var module = new DependencyTrackingTelemetryModule())
+            {
+                module.ProfileQueryEndpoint = FakeProfileApiEndpoint;
+                module.Initialize(this.config);
+                Assert.IsTrue(DependencyTableStore.IsDesktopHttpDiagnosticSourceActivated);
+
+                var url = new Uri(LocalhostUrl + "123?q=123");
+                HttpWebRequest request = WebRequest.CreateHttp(url);
+
+                using (new LocalServer(LocalhostUrl))
+                {
+                    using (request.GetResponse())
+                    {
+                    }
+                }
+
+                this.ValidateTelemetryForDiagnosticSource(this.sentTelemetry.Single(), url, request, true, "200");
+            }
+        }
+
+        [TestMethod]
+        [Timeout(5000)]
+        public void TestHttpRequestsWithQueryStringAreCollectedEventSource()
+        {
+            using (var module = new DependencyTrackingTelemetryModule())
+            {
+                module.DisableDiagnosticSourceInstrumentation = true;
+                module.ProfileQueryEndpoint = FakeProfileApiEndpoint;
+                module.Initialize(this.config);
+
+                var url = new Uri(LocalhostUrl + "123?q=123");
+                HttpWebRequest request = WebRequest.CreateHttp(url);
+
+                using (new LocalServer(LocalhostUrl))
+                {
+                    using (request.GetResponse())
+                    {
+                    }
+                }
+
+                this.ValidateTelemetryForEventSource(this.sentTelemetry.Single(), url, request, true, "200");
```

Comment:
I see your point now.
From one side I agree with you and if it were production code it would for sure make sense.

From the other side I think bringing more logic into unit tests does not pay-off and complicates things for someone would debug it one year later.
It would have make sense to have parameterized tests with simple switch between eventsource/ diagnosticsource, however mstest does not have good support for it and it would be difficult to understand which test has failed (we already have this problem for NET40/NET45 tests).

So I'm hesitant to make this change. But If you really want it and think it makes sense, I will do it.